homeassistant/components/sensor/__init__.py

```diff
@@ -546,6 +546,87 @@ class SensorStateClass(StrEnum):
     SensorDeviceClass.WIND_SPEED: set(UnitOfSpeed),
 }
 
+DEVICE_CLASS_STATE_CLASSES: dict[SensorDeviceClass, set[SensorStateClass | None]] = {
+    SensorDeviceClass.APPARENT_POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.AQI: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ATMOSPHERIC_PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.BATTERY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CO: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CO2: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CURRENT: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.DATA_RATE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.DATA_SIZE: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.DATE: {None},
+    SensorDeviceClass.DISTANCE: {None}.union(set(SensorStateClass)),
+    SensorDeviceClass.DURATION: {None},
+    SensorDeviceClass.ENERGY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.ENUM: {None},
+    SensorDeviceClass.FREQUENCY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.GAS: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.HUMIDITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ILLUMINANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.IRRADIANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MOISTURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MONETARY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.NITROGEN_DIOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROGEN_MONOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROUS_OXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.OZONE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM1: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM10: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM25: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER_FACTOR: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PRECIPITATION: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
```

Comment:
Maybe here also. I think in terms of "since T" it can only go up, but in terms of "in last X hours/mins" it can go either up or down.

Reply:
Not sure about this one tbh. I'll remove the rule and allow for all.

Reply:
I think your original call was the correct one. If the state is a MEASUREMENT, the device class would be PRECIPITATION_INTENSITY.